apps/local_laplacian/local_laplacian_generator.cpp

```diff
@@ -124,17 +117,16 @@ class LocalLaplacian : public Halide::Generator<LocalLaplacian> {
             }
             outGPyramid[0]
                 .compute_at(output, y).vectorize(x, 8);
-            for (int j = 5; j < J; j++) {
+            for (int j = 5; j < pyramid_levels; j++) {
                 inGPyramid[j].compute_root();
                 gPyramid[j].compute_root().parallel(k);
                 outGPyramid[j].compute_root();
             }
         }
-
-        return output;
     }
 private:
     Var x, y, c, k;
+    Func remap, gray, inGPyramid[maxJ], gPyramid[maxJ], outGPyramid[maxJ];
```

Comment:
> You shouldn't have to be aware of a feature until it solves a problem that you have

That's a good principle and one I should take to heart here. It's definitely a defect in the current design. It's no secret that the current generate()/schedule() proposal isn't really proving to be an appealing solution to the problem it intended to solve. (I've been looking over the auto_schedule branch last night for the first time in many months and realizing that it might not be a good fit there, either, but need to ponder and experiment some more.)

> The point of this is so that you can do all your scheduling at the bottom of generate(), and there's no schedule() method.
  
Side note: even if we go down the path of nuking the schedule() method, I think that moving towards Input<>/Output<>/generate() is a desirable goal, if for no other reason than forcing the Outputs to be enumerated and typed make for easier-to-read code.

> So what if we allowed scheduling to use unbound ScheduleParams?

This is an interesting idea. I'll try to explore it in a side branch today.

> I realize you're ill

Responding intermittently between naps and tea :-)